internal/services/filehash/filesystemhasher.go

```diff
@@ -0,0 +1,102 @@
+package filehash
+
+import (
+	"context"
+
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/utils/fshasher"
+	"github.com/sirupsen/logrus"
+)
+
+type FileHash struct {
+	Path string
+	Hash string
+}
+
+type FileSystemHasher struct {
+	log     *logrus.Logger
+	alg     string
+	workers int
+}
+
+func NewFileSystemHasher(log *logrus.Logger, alg string, workers int) *FileSystemHasher {
+	return &FileSystemHasher{
+		log:     log,
+		alg:     alg,
+		workers: workers,
+	}
+}
+
+// Calculate calculate file hashes synchronously and store into slice
+func (fhs *FileSystemHasher) CalculateAll(ctx context.Context, dirPath string) ([]FileHash, error) {
+	hashChan := make(chan FileHash)
+	result := make([]FileHash, 0, 1024)
+	var err error
+
+	go func() {
+		fhs.log.Debug("Begin calculate hashes")
+		err = fshasher.Walk(ctx, fhs.log, fhs.workers, dirPath, fhs.alg, func(filePath string, fileHash string) error {
+			hashChan <- FileHash{Path: filePath, Hash: fileHash}
+			return nil
+		})
+		close(hashChan)
+	}()
+
+	for h := range hashChan {
+		result = append(result, h)
+	}
+
+	if err != nil {
+		fhs.log.WithError(err).Debug("Failed calculate hashes")
+		return nil, err
+	}
+
+	fhs.log.WithField("HashesCount", len(result)).Debug("Success calculate hashes")
+
+	return result, nil
+}
+
+// CalculateInCallback calculate file hashes and call callback for each hash
+func (fhs *FileSystemHasher) CalculateInCallback(ctx context.Context, dirPath string, handlert func(fh FileHash) error) error {
```

Comment:
It seems that this func is not used

Reply:
I wanted to use this function to compare calculated hash and stored in db (now we aggregate all hashes and compare with all hashes in from db).
But we already decided to use pipes in future